### PR TITLE
Refactor php class structure and include the new autoloader

### PR DESCRIPTION
**Pull request by mikejolley**
This is a continuation of https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/628 in which all PHP code is moved into the `src/` directory with proper namespaces and autoloading.

This now uses the `jetpack-autoloader` library which is supported in my branch on core here: https://github.com/woocommerce/woocommerce/pull/23957

Blocks are found in `src/BlockTypes` and handle registration and render. There are abstract classes for the GRID type and Blocks in general.

The main plugin file will never be loaded by WooCommerce core. Core will load the Library straight up during init.

Since we cannot build files before pushing to Packagist like with NPM, I've included a deploy script which runs build, creates a branch, force pushes `/build` and creates a release/tag all automatically. This can be ran/tested with:

```bash
npm run deploy
```

If you do this, use a test version such as `2.2.0-dev`, just ensure it does not already exist. Example: https://github.com/woocommerce/woocommerce-gutenberg-products-block/releases/tag/v2.2.0-dev

In the dist version, all webpack config and package.json files are skipped since the package is pre-built. This will be the version imported to core.

### CODE REVIEW

File: src/Assets.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+			),
+			'stockStatuses' => wc_get_product_stock_status_options(),
+			'siteTitle'     => get_bloginfo( 'name' ),
+			'dataEndpoints' => array(),
+			'l10n'          => array(
+				'userLocale'    => get_user_locale(),
+				'weekdaysShort' => array_values( $wp_locale->weekday_abbrev ),
+			),
+		);
+		// NOTE: wcSettings is not used directly, it's only for @woocommerce/components.
```
Comment (timmyc) [inline src/Assets.php]: I don't think this comment is applicable here, but having it above makes sense.

File: src/Assets.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	 * based on user-controlled settings from WordPress.
+	 *
+	 * @since 2.0.0
+	 */
+	public function print_script_settings() {
+		global $wp_locale;
+		$code           = get_woocommerce_currency();
+		$product_counts = wp_count_posts( 'product' );
+
+		// NOTE: wcSettings is not used directly, it's only for @woocommerce/components
```
Comment (timmyc) [inline src/Assets.php]: Wondering if we should add a TODO here to remove this once wc-admin is in core too... @ryelle thoughts?

**Comment by mikejolley** [comment]
Thanks @timmyc Ill take a look into the error. The 2 comments were just moved, not added so may be out of scope and can be updated in another commit.

I'm going to make some final changes to this branch to include the files the new core PackageManager class will need, and then I'll test this works when included via Packagist :)

@ryelle I spoke to @claudiosanches about builds of files and he seemed happy with the build being ran after the `composer install` within the Woo core deploy script. That means we don't need build files in source control or on Packagist. Makes this much simpler tbh.

**Comment by mikejolley** [comment]
> The only oddity I'm seeing is if I create blocks using this branch of the plugin, and then deactivate the plugin, so defaulting to the versions that are in Woo core, my post with blocks in it looks like

@timmyc This is forked from master so there are new block structures in place. The version in core isn't compatible. That's not a problem afaik.

**Comment by mikejolley** [comment]
I'm going to break this up and do some follow up PRs since this is out of sync with master anyway. I'll use git mv to preserve history on the block classes.

**Pull request closed without merging**